runtime/drivers/repo.go

```diff
@@ -28,6 +30,11 @@ type RepoStore interface {
 	Watch(ctx context.Context, cb WatchCallback) error
 }
 
+type RepoHealth struct {
+	Sync         error
+	AdminConnect error
+}
```

Comment:
Feels like this leaks a specific connector (`admin`) to all the other connectors. I would suggest keeping it simple by relying on a `Ping(ctx) error` on the handle itself. The `admin` connector can use `errors.Join` to combine the sync and ping error.

From the health check endpoint side, it's okay just to have that one error for the admin connector.

Reply:
The idea behind not combining errors was that admin service(or manual) can take decision what to do based on different parameters. 
A `sync` error may just require triggering a reconcile while a `connect` error will require `reset`/`redeploy`. I am wondering what action will be taken in case we combine these errors ?

Reply:
> Feels like this leaks a specific connector (admin) to all the other connectors. 

Only to connectors implementing `RepoStore`. Can also be changed to just `Connect`.